### ISSUE TITLE
WP_HOME and WP_SITEURL when going live.

### ISSUE THREAD

**Issue by alexfornuto** [pantheon-systems/documentation#2405]
Re: <https://pantheon.io/docs/going-live/>
## Issue Description
 - UPDATE: With our new redirect snippet, we take care of these values, however in doing so a PHP notice is thrown because our upstream defines both constants in wp-config.php
 - Based on ticket 9442346850
 - Customer says that the need to configure the `WP_HOME` and `WP_SITEURL` values in `wp-config.php` are not clearly documented in the going live process.

## Suggested Resolution
 - Add (or move for higher visibility) information on setting these values for a custom domain name.

**Comment by alexfornuto** [comment]
@joshkoenig @greg-1-anderson @danielbachhuber: do any of you have any input on this?

**Comment by danielbachhuber** [comment]
I'm not confident in the specifics, but I suspect @ataylorme would know the answer.

**Comment by ataylorme** [comment]
@alexfornuto,
`WP_HOME` and `WP_SITEURL` are defined in [Pantheon's default `wp-config.php`](https://github.com/pantheon-systems/WordPress/blob/master/wp-config.php#L78:L79).

They are set to the current URL for a few reasons:
* The settings can be the same for all environments (dev, test, live, multidev)
* Users may have multiple custom domains added to a site and the snippet this way allows all of them to resolve

If users only have one custom domain and wish to force that domain for live they can do so by replacing those two lines with something like:
```
if( 'live' === $_ENV['PANTHEON_ENVIRONMENT'] ){
    define('WP_HOME', $scheme . '://' . 'my-custom-domain.com');
    define('WP_SITEURL', $scheme . '://' . 'my-custom-domain.com');
} else {
    define('WP_HOME', $scheme . '://' . $_SERVER['HTTP_HOST']);
    define('WP_SITEURL', $scheme . '://' . $_SERVER['HTTP_HOST']);
}
```

We can't recommend this for all sites or use it as a default because some sites have multiple domains but I think this would be good to document as a suggestion for users with one domain.

Josh/Greg I'd like your thoughts as well

**Comment by alexfornuto** [comment]
@joshkoenig @greg-1-anderson Tagging you since @ataylorme asked for input without tagging.

**Comment by ttrowell** [comment]
@alexfornuto I think this issue might be mute based on the current redirect snippets here:

https://pantheon.io/docs/guides/launch/redirects/

**Comment by alexfornuto** [comment]
From another issue on this subject:

>Change it to a variable, don't redefine again, or we can have more specific instructions on how to use the snippet.

**Comment by dustinleblanc** [comment]
The original issue from the chat could have likely been solved by a search and replace. When customers clone the database to live and subsequently add a custom domain, we don't perform a search and replace on the domain. If they add the custom domain, then clone the database to live, this doesn't happen.

Hardcoding the constants is a work around that is sometimes useful for sites with edge cases, the same way some odd sites work better with `$base_url` defined in settings.php, its a site by site, case by case basis. While it might be good to document some of these snippets (they are fairly common and can follow a pattern), I don't think it should be part of the going live documentation.

What might be useful here though is a disclaimer about launching Wordpress sites and the order of operations (cloning, then adding domain == bad time).

**Comment by alexfornuto** [comment]
closed by #2853
